base/namedtuple.jl

```diff
@@ -35,7 +45,7 @@ end
 convert(::Type{NamedTuple{names,T}}, itr::NamedTuple{names,T}) where {names,T} = itr
 
 function convert(::Type{NamedTuple{names,T}}, itr) where {names,T}
-    namedtuple(NamedTuple{names}, T(itr)...)
+    namedtuple(NamedTuple{names,T}, T(itr)...)
```

Comment:
I find it very unfortunate that (a) this method assumes that the names from iter are insignificant (b) this method assumes that iterables can be converted to a Tuple by calling the Tuple constructor

Reply:
> this method assumes that iterables can be converted to a Tuple by calling the Tuple constructor

When is that not the case?

I do agree this shouldn't be a method of `convert` though. Most likely named tuples should only be `convert`able to other named tuple types.